pkg/certificate/authorization/endpoint.go

```diff
@@ -0,0 +1,141 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package authorization
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"storj.io/storj/pkg/pb"
+)
+
+// ErrEndpoint is the default error class for the authorization endpoint.
+var ErrEndpoint = errs.Class("authorization endpoint error")
+
+// Endpoint implements pb.AuthorizationsServer
+type Endpoint struct {
+	log      *zap.Logger
+	db       *DB
+	server   http.Server
+	listener net.Listener
+}
+
+// NewEndpoint creates a new authorization gRPC server.
+func NewEndpoint(log *zap.Logger, db *DB, listener net.Listener) *Endpoint {
+	mux := http.NewServeMux()
+	endpoint := &Endpoint{
+		log:      log,
+		db:       db,
+		listener: listener,
+		server: http.Server{
+			Handler: mux,
+		},
+	}
+
+	mux.HandleFunc("/v1/authorization/create", endpoint.httpCreate)
+
+	return endpoint
+}
+
+// Create creates an authorization from the given authorization request.
+func (endpoint *Endpoint) Create(ctx context.Context, req *pb.AuthorizationRequest) (_ *pb.AuthorizationResponse, err error) {
+	mon.Task()(&ctx, req.UserId)(&err)
+
+	group, err := endpoint.db.Create(ctx, req.UserId, 1)
+	if err != nil {
+		msg := "error creating authorization"
+		err := ErrEndpoint.New(msg)
+		endpoint.log.Error(msg, zap.Error(err))
+		return nil, status.Error(codes.Internal, err.Error())
```

Comment:
why not:
```go
endpoint.log.Error(msg, zap.Error(ErrEndpoint.New(msg)))
return nil, status.Error(codes.Internal, msg)
```

Reply:
We don't get the `ErrEndpoint` wrapping in the grpc error if we do that.

Reply:
hmm result for endpoint consumer will be `authorization endpoint error: error creating authorization`, ok looks good. One thing that bothers me a little bit is `endpoint` in the default error message. I may give an impression that something goes wrong with endpoint not the service itself but treat that as nit.

Reply:
What are you calling the "service" here?

Reply:
because by definition endpoint shouldn't contain any business logic, its just processing network requests. Here we don't have explicity different service but still, endpoint is just passing a request to `DB` and error is coming from there. 
https://github.com/storj/storj/wiki/Architecture#layers